src/main/resources/googlepubsub-projectsSubscriptions/createTopicSubscription.xml

```diff
@@ -0,0 +1,91 @@
+    <sequence>
+        <property name="uri.var.googlepubsub.projectId" expression="$func:projectId"/>
+        <property name="uri.var.googlepubsub.subscriptionName" expression="$func:subscriptionName"/>
+        <property name="googlepubsub.topicName" expression="$func:topicName"/>
+        <property name="googlepubsub.ackDeadlineSeconds" expression="$func:ackDeadlineSeconds"/>
+        <property name="googlepubsub.pushEndpoint" expression="$func:pushEndpoint"/>
+        <property name="googlepubsub.attributes" expression="$func:attributes"/>
+        <payloadFactory media-type="json">
+            <format>
+                {
+                "topic": "$1"
+                }
+            </format>
+            <args>
+                <arg expression="fn:concat('projects/',$ctx:uri.var.googlepubsub.projectId,'/topics/',$ctx:googlepubsub.topicName)"/>
+            </args>
+        </payloadFactory>
+        <script language="js">
+            <![CDATA[
+                var ackDeadlineSeconds = mc.getProperty('googlepubsub.ackDeadlineSeconds');
+                var pushEndpoint = mc.getProperty('googlepubsub.pushEndpoint');
+                var attributes = mc.getProperty('googlepubsub.attributes');
+                var payload = mc.getPayloadJSON();
+                if (ackDeadlineSeconds != null && ackDeadlineSeconds != "") {
+                    payload.ackDeadlineSeconds = ackDeadlineSeconds;
+                }
+                if (pushEndpoint != null && pushEndpoint != "") {
+                    payload.pushConfig={};
+                    payload.pushConfig.pushEndpoint=pushEndpoint;
+                    if (attributes != null && attributes != "") {
+                        payload.pushConfig.attributes = attributes;
+                    }
+                }
+                mc.setPayloadJSON(payload);
+            ]]>
+        </script>
+        <property name="Accept-Encoding" scope="transport" action="remove"/>
+        <filter source="$ctx:googlepubsub.blocking" regex="true">
+            <then>
+                <call blocking="true">
+                    <endpoint>
+                        <http method="put"
+                              uri-template="{uri.var.googlepubsub.apiUrl}/{uri.var.googlepubsub.apiVersion}/projects/{uri.var.googlepubsub.projectId}/subscriptions/{uri.var.googlepubsub.subscriptionName}"/>
+                    </endpoint>
+                </call>
+            </then>
+            <else>
+                <call>
+                    <endpoint>
+                        <http method="put"
+                              uri-template="{uri.var.googlepubsub.apiUrl}/{uri.var.googlepubsub.apiVersion}/projects/{uri.var.googlepubsub.projectId}/subscriptions/{uri.var.googlepubsub.subscriptionName}"/>
+                    </endpoint>
+                </call>
+            </else>
+        </filter>
+        <!-- Remove response custom header information -->
```

Comment:
Why do we remove all header information? We only have to remove API-specific headers, isn't it?

Reply:
Yeah, We don't need remove all header information.We only have to remove API-specific headers.